README.md

```diff
@@ -94,6 +97,39 @@ An additional dimension that is not `spatial`, but may be `temporal` if the data
 
 An Additional Dimension Object MUST specify an `extent` or a set of `values`. It MAY specify both.
 
+### Variable Object
+
+A *Variable Object* defines a variable (or a multi-dimensional array). The variable may have dimensions, which are described by [Dimension Objects](#dimension-object).
+
+| Field Name       | Type                 | Description |
+| ---------------- | ---------------------| ----------- |
+| dimensions       | \[string]            | **REQUIRED.** The dimensions of the variable. This should refer to keys in the ``cube:dimensions`` object or be an empty list if the variable has no dimensions. |
+| type             | string               | **REQUIRED.** Type of the variable, (`data`, `auxiliary`). |
+| description      | string               | Detailed multi-line description to explain the dimension. [CommonMark 0.29](http://commonmark.org/) syntax MAY be used for rich text representation. |
+| extent           | \[number\|null]      | If the dimension consists of [ordinal](https://en.wikipedia.org/wiki/Level_of_measurement#Ordinal_scale) values, the extent (lower and upper bounds) of the values as two-dimensional array. Use `null` for open intervals. |
+| values           | \[number\|string]    | A set of all potential values, especially useful for [nominal](https://en.wikipedia.org/wiki/Level_of_measurement#Nominal_level) values. |
+| step             | number\|null         | If the dimension consists of [interval](https://en.wikipedia.org/wiki/Level_of_measurement#Interval_scale) values, the space between the values. Use `null` for irregularly spaced steps. |
+| unit             | string               | The unit of measurement for the data, preferably compliant to [UDUNITS-2](https://ncics.org/portfolio/other-resources/udunits2/) units (singular). |
+
+**type**: The Variable `type` indicates whether what kind of variable is being described,
+using the terminology from the [CF Conventions](http://cfconventions.org/Data/cf-conventions/cf-conventions-1.8/cf-conventions.html#terminology).
+A `data` variable typically represents some physical quantity being measured.
+
+An `auxiliary` coordinate variable is defined as
+
+> Any netCDF variable that contains coordinate data, but is not a coordinate variable
```

Comment:
Could we write this in a way that it's easier to understand and less specific to netCDF? For example, I have no clue what a NUG is.

As only auxiliary and cooridnate is allowed, maybe put coordinate first (it's easier to understand) and then say auxiliary is everything else? Could there be any other cases not covered by and of the types?